polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -203,29 +211,38 @@ export function Item({
     );
   }
 
-  const secondaryActionLinkMarkup = secondaryAction && (
-    <UnstyledLink
-      external
-      url={secondaryAction.url}
-      className={styles.SecondaryAction}
-      tabIndex={tabIndex}
-      aria-disabled={disabled}
-      aria-label={secondaryAction.accessibilityLabel}
-      onClick={secondaryAction.onClick}
-    >
-      <Icon source={secondaryAction.icon} />
-    </UnstyledLink>
-  );
+  if (secondaryAction && process.env.NODE_ENV === 'development') {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'Deprecation: The `secondaryAction` prop on the `Navigation.Item` has been deprecated. Use `secondaryActions` instead.',
+    );
+  }
+
+  const actions = secondaryActions || (secondaryAction && [secondaryAction]);
+
+  if (actions && actions.length > MAX_SECONDARY_ACTIONS) {
+    actions.length = MAX_SECONDARY_ACTIONS;
```

Comment:
oh nice! This seems more efficient than slicing out everything after the 2nd item